File: 0704-binary-search/0704-binary-search.dart
class Solution {
  int search(List<int> nums, int target) {
    int answer = -1;
    int start = 0;
    int end = nums.length - 1;
    while (end - start > 1) {
        int pivot = start + (end - start) ~/ 2;
        if (nums[pivot] == target) {
            return pivot;
        } else if (nums[pivot] > target) {
            end = pivot;
        } else {
            start = pivot;
        }
    }
    if (nums[start] == target) {
        return start;
    } else if (nums[end] == target) {
        return end;
    }
    
    return answer;
  }
}

File: 0704-binary-search/0704-binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        answer = -1
        start = 0
        end = len(nums) - 1
        while end - start > 1:
            pivot = start + (end - start) // 2
            if nums[pivot] == target:
                return pivot
            elif nums[pivot] > target:
                end = pivot
            else:
                start = pivot
        if nums[start] == target:
            return start
        if nums[end] == target:
            return end
        return answer


